### PR TITLE
Start internet stream service by default. It runs once per day. Fix b…

### PR DESCRIPTION
**Pull request by david-developer-play**
…ug in internet stream service. Fix typo in configuration url. Add more realistic testing data to demo_dump.

Update test, update delimiter and resources to comply with new format.

### CODE REVIEW

File: server-kotlin/src/main/resources/application.properties [hunk 1]
```diff
@@ -17,8 +17,15 @@ city-request.appName=CityVizor
...
 # Fulltext search
 fulltextSearch.indexRefreshEnabled=true
-# How often service which downloads budgets runs
-internet.stream.service.configuration.frequency=PT60M
+# How often service which downloads budgets runs, once per day
+internet.stream.service.configuration.frequency=PT1440M
 # Initial delay before service starts, this ensures that service is configured
 # and connected to db
 internet.stream.service.configuration.initDelay=PT5M
+# map of configuration, key is url of city as defined in app.profiles, column url
```
Comment (lukas-krecan) [inline server-kotlin/src/main/resources/application.properties:25]: I would prefer to not start it by default. We can move it to a separate properties file (application-citysync.properties) and include it using Spring profiles (https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-profiles)

Comment (lukas-krecan) [reply server-kotlin/src/main/resources/application.properties:25]: Mimochodem, proč to chceš zapínat všude?

File: server-kotlin/src/main/kotlin/digital/cesko/internet_stream/InternetStreamService.kt [hunk 2]
```diff
@@ -47,6 +50,7 @@ class InternetStreamService(
         fixedRateString = "\${internet.stream.service.configuration.frequency}",
         initialDelayString = "\${internet.stream.service.configuration.initDelay}"
     )
+    @Profile("internetstream")
```
Comment (lukas-krecan) [inline server-kotlin/src/main/kotlin/digital/cesko/internet_stream/InternetStreamService.kt:53]: Tady to neni potreba

File: server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt [hunk 3]
```diff
@@ -6,25 +6,28 @@ import org.jetbrains.exposed.sql.select
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Profile
 import java.math.BigDecimal
 import kotlin.test.assertEquals
 
+@Profile("internetstream")
```
Comment (lukas-krecan) [inline server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt:13]: Tady to neni nutne

File: server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt [hunk 4]
```diff
@@ -6,25 +6,28 @@ import org.jetbrains.exposed.sql.select
...
 class InternetStreamServiceTest : AbstractSpringDatabaseTest() {
     @Autowired
     private lateinit var internetStreamService: InternetStreamService
 
     @Test
+    @Profile("internetstream")
```
Comment (lukas-krecan) [inline server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt:19]: Tady taky ne, v testech je to zapnute vzdy

File: server-kotlin/src/main/resources/application.properties [hunk 5]
```diff
@@ -17,8 +17,6 @@ city-request.appName=CityVizor
...
 # Fulltext search
 fulltextSearch.indexRefreshEnabled=true
-# How often service which downloads budgets runs
-internet.stream.service.configuration.frequency=PT60M
-# Initial delay before service starts, this ensures that service is configured
-# and connected to db
-internet.stream.service.configuration.initDelay=PT5M
+# If set to internetstream, run service which synchronizes data from internet stream
+# Tge service does not run if set to any other value
+spring.profiles.active=default
```
Comment (lukas-krecan) [inline server-kotlin/src/main/resources/application.properties:22]: Tohle neni potreba, default je default

File: server-kotlin/src/test/resources/application-test.properties [hunk 6]
```diff
@@ -4,8 +4,5 @@ city-request.listName=Sheet1
 city-request.sheetId=1f6LnMH7w5Up8AJlTYC0x0AOYpRVLaT62KvW3GWF9tr8
 googleCredentials=classpath:city_request/test-credentials.json
 fulltextSearch.indexRefresh.delay=PT1H
-internet.stream.service.configuration.urls.mestocernosice=classpath:internet_stream/
-internet.stream.service.configuration.urls.ub=classpath:internet_stream/
-internet.stream.service.configuration.urls.nmnm=classpath:internet_stream/
-# test file with example data
-internet.stream.service.configuration.fileUrls=test_CSV.zip
+# active for testing
+spring.profiles.active=internetstream
```
Comment (lukas-krecan) [inline server-kotlin/src/test/resources/application-test.properties:8]: Tady se da pouzit `spring.profiles.include`

**Comment by srdecny** [comment]
I'm not familiar with Spring/Kotlin in general, but is there a way to toggle/configure the InternetStream service without restarting the `server-kotlin` ? Because right now it seems like I have to manually edit the `server-kotlin/src/main/resources/application-internetstream.properties` file and then restart the server, is that right?

**Comment by lukas-krecan** [comment]
> I'm not familiar with Spring/Kotlin in general, but is there a way to toggle/configure the InternetStream service without restarting the `server-kotlin` ? Because right now it seems like I have to manually edit the `server-kotlin/src/main/resources/application-internetstream.properties` file and then restart the server, is that right?

That's right. In theory it's possible to switch it in the runtime but there is rarely need for it. You either want the config in given environment or not.

**Comment by srdecny** [comment]
I'm not sure if I like that. For example, Prague (or any other interested  runs it's own instance of CityVizor, and I don't think they're interested in the InternetStream functionality at all, so allowing the InternetStream service to be enabled/disabled via an environment variable (for example) would be useful. Right now to disable InternetStream, after every pull of a new image, they would have to manually edit the config file + restart the kotlin server in the container every time, which isn't practical at all.

**Comment by lukas-krecan** [comment]
> I'm not sure if I like that. For example, Prague (or any other interested city that wants to self-host) runs it's own instance of CityVizor, and I don't think they're interested in the InternetStream functionality at all, so allowing the InternetStream service to be enabled/disabled via an environment variable (for example) would be useful. Right now to disable InternetStream, after every pull of a new image, they would have to manually edit the config file + restart the kotlin server in the container every time, which isn't practical at all.

I see. You can of course use environment variables to override whatever you want. So you can provide SPRING_PROFILE_INCLUDE env variable and add internetstream profile. Or you can specify individual properties if you need to (we are using the same tool to specify DB credentails)

**Comment by srdecny** [comment]
Ah, okay, that's sensible then. No further objections :)

File: server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt [hunk 7]
```diff
@@ -6,9 +6,11 @@ import org.jetbrains.exposed.sql.select
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.context.ActiveProfiles
 import java.math.BigDecimal
 import kotlin.test.assertEquals
 
+@ActiveProfiles("internetstream")
 class InternetStreamServiceTest : AbstractSpringDatabaseTest() {
```
Comment (lukas-krecan) [inline server-kotlin/src/test/kotlin/digital/cesko/internet_stream/InternetStreamServiceTest.kt:14]: Staci to bud v properties souboru nebo v ActiveProfiles, ale neskodi to

**Review by lukas-krecan — APPROVED** [review]

**Pull request merged**